tools/include/eosio/codegen.hpp

```diff
@@ -265,16 +281,173 @@ namespace eosio { namespace cdt {
+            if (func_decl->isThisDeclarationADefinition() && func_decl->hasBody()) {
+               Stmt *stmts = func_decl->getBody();
+               for (auto it = stmts->child_begin(); it != stmts->child_end(); ++it) {
+                  if (Stmt *s = *it) {
+                     if (auto* ec = dyn_cast<ExprWithCleanups>(s)) {
+                        s = ec->getSubExpr();
+                        while (auto* ice = dyn_cast<ImplicitCastExpr>(s))
+                           s = ice->getSubExpr();
+                     }
+
+                     if (auto* call = dyn_cast<CallExpr>(s)) {
+                        if (FunctionDecl *fd = call->getDirectCallee()) {
+                           if (func_calls.count(fd) == 0) {
+                              process_function(fd);
+                           }
+                           if (!func_calls[fd].empty()) {
+                              func_calls[func_decl].push_back(call);
+                              break;
+                           }
+                        } else {
+                           process_indi_callee(func_decl, call);
+                        }
+                     } else if (auto* ds = dyn_cast<DeclStmt>(s)) {
+                        auto process_decl = [this]( DeclStmt *s ) {
+                           for (auto it = s->decl_begin(); it != s->decl_end(); ++it) {
+                              if (auto* vd = dyn_cast<VarDecl>(*it)) {
+                                 if (Expr *init = vd->getInit()) {
+                                    process_decl_init(vd, init);
+                                 }
+                              }
+                           }
+                        };
+                        process_decl(ds);
+                     } else if (auto* bo = dyn_cast<BinaryOperator>(s)) {
+                        auto process_assignment = [this]( BinaryOperator *b ) {
+                           Expr *lhs = nullptr, *rhs = nullptr;
+                           if ((lhs = b->getLHS()) && (rhs = b->getRHS())) {
+                              if (FunctionDecl *fd = get_rhs_fd(rhs)) {
+                                 if (auto* lhs_dre = dyn_cast<DeclRefExpr>(lhs)) {
+                                    update_indi_func_map(lhs_dre->getFoundDecl(), fd);
+                                 } else if (auto* lhs_me = dyn_cast<MemberExpr>(lhs)) {
+                                    update_indi_func_map(lhs_me->getMemberDecl(), fd);
+                                 }
+                              }
+                           }
+                        };
+                        process_assignment(bo);
+                     }
+                  }
+               }
+            }
+         }
+
+         virtual bool VisitFunctionDecl(FunctionDecl* func_decl) {
+            SourceManager &sm = get_rewriter().getSourceMgr();
+            if (sm.isInSystemHeader(func_decl->getLocation()) || sm.isInExternCSystemHeader(func_decl->getLocation())) {
```

Comment:
This is a fine short circuit.  But, we do have a few places in our C++ and C headers where we have some `extern C <writable_host_functions>` to supply the underlying bases for libc.  This will more than likely miss those calls.

Reply:
This system header checking is deleted in the new commit: https://github.com/EOSIO/eosio.cdt/blob/ef00c9d035fcea306131884724691398189be4a3/tools/include/eosio/codegen.hpp#L383-L384